diff -aurN a/bootstrap.sh b/bootstrap.sh
--- a/bootstrap.sh	2013-03-09 14:55:05 -0800
+++ b/bootstrap.sh	2013-07-18 00:12:38 -0700
@@ -215,7 +215,7 @@
 if test "x$BJAM" = x; then
   echo -n "Building Boost.Build engine with toolset $TOOLSET... "
   pwd=`pwd`
-  (cd "$my_dir/tools/build/src/engine" && ./build.sh "$TOOLSET") > bootstrap.log 2>&1
+  (cd "$my_dir/tools/build/src/engine" && ./build.sh mingw) > bootstrap.log 2>&1
   if [ $? -ne 0 ]; then
       echo
       echo "Failed to build Boost.Build build engine"
